tests/browser/custom-reporter.ts

```diff
@@ -47,10 +74,14 @@ class MyReporter implements Reporter {
         break;
     }
 
-    console.log(`${hierarchyPath} > ${test.title}: ${result.status}`);
+    console.log(
+      `[${this.currentTestIndex}/${this.totalTests}] ${hierarchyPath} > ${test.title}: ${result.status}`
+    );
   }
 
   onEnd(result: FullResult) {
+    if (this.isListMode) return;
```

Comment:
I don't know how much work it would be, but the thing I found most useful in the `--list` feature of the default reporter was telling me how many tests would be run. It looks liket his:

<img width="287" alt="Screenshot 2024-12-23 at 11 11 59 AM" src="https://github.com/user-attachments/assets/86ae488d-6e33-4f68-b30d-e9bb0cb983d1" />

I use the list command to figure out how many tests are run in the smoke tests vs the nightly tests, how many actual test variations are executed for a given change (all the permutations of a story with themes and browsers), etc.

Reply:
Thanks, that's good to know! I've added the total number of tests and test files to the summary output at the end, matching the default reporter's format. 

Are the permutations of story, theme, and browser readable enough? It looks to me like we're listing everything the same as the default reporter, but let me know if I've missed any nuances.

Reply:
This is great!